FreeRTOS/FreeRTOS-Kernel
# fixed deadlock in event groups when a mutex is used for memory alloca…

…tion

Deadlock in event groups when a mutex is used for memory allocation

Description
-----------
If the pvPortMalloc and vPortFree functions are protected with a mutex and the vPortFree is called in a vTaskSuspendAll section
(which was the case) it causes a deadlock when a task with lower priority locked the memory allocation mutex.

I'm not sure if this is an allowed scenario but we had the problem that our high priority tasks where blocked for up to 23ms because of the vTaskSuspendAll in the memory allocation. To prevent that we used a mutex to protect the memory allocation but then we found this problem.

Test Steps
-----------
Allocate Memory with a task with priority 2 ensure that there's a task switch during the allocation to a higher priority task which calls vEventGroupDelete()

Related Issue
-----------
No Issue


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
